### PR TITLE
DB matrix for Github Actions

### PR DESCRIPTION
**Pull request by cedric-anne**

### CODE REVIEW

File: .github/workflows/ci-linux.yml [hunk 1]
```diff
@@ -16,55 +16,44 @@ jobs:
 
     strategy:
       matrix:
-        php-versions: [ '7.3', '8.0', '8.1' ]
-        coverage: [none]
+        php-versions: [ '7.3', '7.4', '8.0', '8.1' ]
+        coverage: ['none']
+        db-image: ['mysql:8.0', 'mariadb:8.0', 'postgres:9.6']
```
Comment (cedric-anne) [inline .github/workflows/ci-linux.yml:21]:
```suggestion
        db-image: ['mysql:8.0', 'mariadb:10.5', 'postgres:9.6']
```

File: .github/workflows/ci-linux.yml [hunk 2]
```diff
@@ -16,55 +16,44 @@ jobs:
...
         ports:
           - 3306:3306
+          - 5432:5432
+        # Set health checks to wait until postgres has started
         options: >-
-          --health-cmd="mysqladmin ping"
+          --health-cmd="if [[ -n $(command -v pg_isready) ]]; then pg_isready; else mysqladmin ping; fi"
           --health-interval=10s
           --health-timeout=5s
           --health-retries=5
```
Comment (cedric-anne) [inline .github/workflows/ci-linux.yml:54]:
```suggestion
          --health-retries=10
```

File: .github/workflows/ci-linux.yml [hunk 3]
```diff
@@ -77,14 +66,12 @@ jobs:
...
       - name: "Show versions"
-        if: env.skip != 'true'
         run: |
           php --version
           composer --version
           echo "node $(node --version)"
           echo "npm $(npm --version)"
-          mysql --version
-          psql --version
+          [ ! "$(docker ps -q -f name=db)" ] || docker exec db bash -c "if [[ -n $(command -v psql) ]]; then psql --version; else mysql --version; fi"
```
Comment (cedric-anne) [inline .github/workflows/ci-linux.yml:74]:
```suggestion
          docker exec ${{ job.services.db.id }} bash -c "if [[ -n $(command -v psql) ]]; then psql --version; else mysql --version; fi"
```

File: .github/workflows/ci-linux.yml [hunk 4]
```diff
@@ -16,55 +16,44 @@ jobs:
...
           MYSQL_ROOT_PASSWORD: g@l3tte
           MYSQL_DATABASE: galette_tests
+        # Open network ports for both mysql and pgsql
         ports:
           - 3306:3306
+          - 5432:5432
+        # Set health checks to wait until postgres has started
         options: >-
-          --health-cmd="mysqladmin ping"
+          --health-cmd="if [[ -n $(command -v pg_isready) ]]; then pg_isready; else mysqladmin ping; fi"
```
Comment (cedric-anne) [inline .github/workflows/ci-linux.yml:51]:
```suggestion
          --health-cmd="if [[ -n $(command -v pg_isready) ]]; then pg_isready -U galette_tests; else mysqladmin ping; fi"
```

File: .github/workflows/ci-linux.yml [hunk 5]
```diff
@@ -77,14 +66,12 @@ jobs:
...
       - name: "Show versions"
-        if: env.skip != 'true'
         run: |
           php --version
           composer --version
           echo "node $(node --version)"
           echo "npm $(npm --version)"
-          mysql --version
-          psql --version
+          docker exec ${{ job.services.db.id }} bash -c "if [[ -n $(command -v psql) ]]; then psql --version; else mysql --version; fi"
```
Comment (cedric-anne) [inline .github/workflows/ci-linux.yml:74]:
```suggestion
          docker exec ${{ job.services.db.id }} bash -c "if [[ -n \$(command -v psql) ]]; then psql --version; else mysql --version; fi"
```

File: .github/workflows/ci-linux.yml [hunk 6]
```diff
@@ -16,55 +16,44 @@ jobs:
...
           MYSQL_ROOT_PASSWORD: g@l3tte
           MYSQL_DATABASE: galette_tests
+        # Open network ports for both mysql and pgsql
         ports:
           - 3306:3306
+          - 5432:5432
+        # Set health checks to wait until postgres has started
         options: >-
-          --health-cmd="mysqladmin ping"
+          --health-cmd="if [[ -n $(command -v pg_isready) ]]; then pg_isready -U galette_tests; else mysqladmin ping; fi"
```
Comment (cedric-anne) [inline .github/workflows/ci-linux.yml:51]:
```suggestion
          --health-cmd="if [[ -n $(command -v pg_isready) ]]; then pg_isready -U galette_tests -d galette_tests; else mysqladmin ping; fi"
```

File: .github/workflows/ci-linux.yml [hunk 7]
```diff
@@ -16,55 +16,44 @@ jobs:
...
           MYSQL_ROOT_PASSWORD: g@l3tte
           MYSQL_DATABASE: galette_tests
+        # Open network ports for both mysql and pgsql
         ports:
           - 3306:3306
+          - 5432:5432
+        # Set health checks to wait until postgres has started
         options: >-
-          --health-cmd="mysqladmin ping"
+          --health-cmd="if [[ -n $(command -v pg_isready) ]]; then pg_isready -U galette_tests -d galette_tests; else mysqladmin ping; fi"
```
Comment (cedric-anne) [inline .github/workflows/ci-linux.yml:51]:
```suggestion
          --health-cmd="bash -c 'if [[ -n $(command -v pg_isready) ]]; then pg_isready; else mysqladmin ping; fi'"
```

**Comment by trasher** [comment]
Thank you very much; it's really great! :+1: 

Current fails on postgres are certainly due to an environment issue; Galette internaly relies on "mysql" or "pgsql" to define the db engine :) I'm not sure about the error with php 7.3 and mysql 8

**Review by trasher — APPROVED** [review]

**Pull request merged**